backend-config/types.go

```diff
@@ -216,11 +243,28 @@ type DgSourceTrackingPlanConfigT struct {
 	TrackingPlan        TrackingPlanT                     `json:"trackingPlan"`
 }
 
+// mergeMaps merges two maps, with values from the second map taking precedence
+func mergeMaps(map1, map2 map[string]interface{}) map[string]interface{} {
+	result := make(map[string]interface{})
+
+	// Copy all key-value pairs from the first map
+	for k, v := range map1 {
+		result[k] = v
+	}
+
+	// Copy all key-value pairs from the second map, overwriting any existing keys
+	for k, v := range map2 {
+		result[k] = v
+	}
+
+	return result
+}
+
 func (dgSourceTPConfigT *DgSourceTrackingPlanConfigT) GetMergedConfig(eventType string) map[string]interface{} {
 	if dgSourceTPConfigT.MergedConfig == nil {
 		globalConfig := dgSourceTPConfigT.fetchEventConfig(GlobalEventType)
 		eventSpecificConfig := dgSourceTPConfigT.fetchEventConfig(eventType)
-		outputConfig := misc.MergeMaps(globalConfig, eventSpecificConfig)
+		outputConfig := mergeMaps(globalConfig, eventSpecificConfig)
```

Comment:
Delete both `mergeMaps` & `misc.MergeMaps`
```suggestion
		outputConfig := lo.Assign(globalConfig, eventSpecificConfig)
```